keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -2025,6 +2006,152 @@ Beta:
+  viewed as a sign that kube-apiserver is rejecting requests, potentially
+  incorrectly. The `apiserver_flowcontrol_request_queue_length_after_enqueue`
+  metric getting too close to the configured queue length could be a sign of
+  insufficient queue size (or a system overload), which can be precursor to
+  rejected requests.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  No. Manual tests will be run before switching feature gate to beta.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+  fields of API types, flags, etc.?** Yes, `--max-requests-inflights` will be
+  deprecated in favor of APF.
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+  If the `apiserver_flowcontrol_dispatched_requests_total` metric is non-zero,
+  this feature is in use. Note that this isn't a workload feature, but a
+  control plane one.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  - [x] Metrics
+    - Metric name: `apiserver_flowcontrol_request_queue_length_after_enqueue`
+    - Components exposing the metric: kube-apiserver
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  No SLOs are proposed for the above SLI.
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?** No.
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+  No.
+
+### Scalability
+
+<!-- I don't have the knowledge to answer some questions in this section. -->
+
+* **Will enabling / using this feature result in any new API calls?** No.
+
+* **Will enabling / using this feature result in introducing new API types?**
+  Yes, a new flowcontrol API group, configuration types, and status types are
+  introduced. See `staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go` for a
+  full list.
+
+* **Will enabling / using this feature result in any new calls to the cloud
+  provider?** No.
+
+* **Will enabling / using this feature result in increasing size or count of
+  the existing API objects?** No.
+
+* **Will enabling / using this feature result in increasing time taken by any
+  operations covered by [existing SLIs/SLOs]?** Yes, a non-negligible latency
+  is added to API calls to kube-apiserver. While [preliminary tests](https://github.com/tkashem/graceful/blob/master/priority-fairness/filter-latency/readme.md)
+  shows that the API server latency is still well within the existing SLOs,
+  more thorough testing needs to be performed.
+
+* **Will enabling / using this feature result in non-negligible increase of
+  resource usage (CPU, RAM, disk, IO, ...) in any components?** The proposed
+  flowcontrol logic in request handling in kube-apiserver will increase the CPU
+  and memory overheads involved in serving each request.
```

Comment:
This question and the last one are a little simplistic, the feature involves both increases and decreases; we hope that it averages out to lower variance / more predictability. More predictability means tighter VM sizing. That is, we hope to make peak load a lower multiple of average load.

Reply:
Also I'd say that users may have to configure the feature to fine-tune it to their use case. The risk is that our default configuration could be a poor fit for some users.

Reply:
Re simplistic - I agree. But going deeper in the question resulted in people not being able to really answer it. So it's a conscious tradeoff :)

Re decreases - this is kind-of implicit, because we hope that we will be able to better tune the apiserver and make it more predictible. But in principle, it adds some cost to processing a single request.
So I actually think the answer is right as is.

> Also I'd say that users may have to configure the feature to fine-tune it to their use case. The risk is that our default configuration could be a poor fit for some users.

It may be not what they need. But in principle it should be better than what they have now (if we do a reasonable job).

Reply:
Added a quick sentence about how users can fine-tune parameters.